main.cpp

```diff
@@ -240,46 +240,87 @@ static int add_endpoint_address(const char *name, size_t name_len, const char *i
     return ret;
 }
 
+static int add_bauds(unsigned long **bauds, size_t *bauds_size, const char *bauds_str)
+{
+    char *tmp, *nul_bauds_str;
+    int count = 1; // we expect at least one baud
+    int i, ret;
+
+    if (!bauds_str || bauds_str[0] == '\0') {
+        *bauds = (unsigned long *)calloc(sizeof(unsigned long), 1);
+        *bauds[0] = DEFAULT_BAUDRATE;
+        *bauds_size = 1;
+        return 0;
+    }
+
+    nul_bauds_str = strdup(bauds_str);
+    if (!nul_bauds_str) {
+        return -errno;
+    }
+
+    for (tmp = nul_bauds_str; *tmp != '\0'; tmp++) {
+        if (*tmp == ',') {
+            count++;
+            *tmp = '\0';
+        }
+    }
+
+    *bauds = (unsigned long *)calloc(sizeof(unsigned long), count);
+    if (!*bauds) {
+        ret = -errno;
+        goto bauds_error;
+    }
+
+    i = 0;
+    *bauds_size = count;
+    tmp = nul_bauds_str;
+    while (count--) {
+        if (safe_atoul(tmp, &(*bauds)[i]) < 0) {
+            log_error("Invalid baud %s", tmp);
+            ret = -EINVAL;
+            goto atoul_error;
+        }
+        i++;
+        tmp += strlen(tmp) + 1;
+    }
+
+    free(nul_bauds_str);
+    return 0;
+
+atoul_error:
+    free(*bauds);
+bauds_error:
+    free(nul_bauds_str);
+    return ret;
+}
+
 static int add_uart_endpoint(const char *name, size_t name_len, const char *uart_device,
-                             unsigned long baudrate)
+                             const char *bauds)
 {
     int ret;
 
     struct endpoint_config *conf
         = (struct endpoint_config *)calloc(1, sizeof(struct endpoint_config));
     assert_or_return(conf, -ENOMEM);
     conf->type = Uart;
-    conf->baud = ULONG_MAX;
 
-    // As name doesn't change, only update if there's no name yet
-    if (!conf->name && name) {
+    if (name) {
         conf->name = strndup(name, name_len);
         if (!conf->name) {
             ret = -ENOMEM;
             goto fail;
         }
     }
 
-    if (uart_device) {
-        free(conf->device);
-        conf->device = strdup(uart_device);
-        if (!conf->device) {
-            ret = -ENOMEM;
-            goto fail;
-        }
-    }
-
+    conf->device = strdup(uart_device);
     if (!conf->device) {
-        ret = -EINVAL;
+        ret = -ENOMEM;
         goto fail;
     }
 
-    if (baudrate != ULONG_MAX) {
-        conf->baud = baudrate;
-    }
-
-    if (conf->baud == ULONG_MAX) {
-        conf->baud = DEFAULT_BAUDRATE;
+    ret = add_bauds(&conf->bauds, &conf->bauds_size, bauds);
```

Comment:
if we are keeping bauds as an array rather than a list, it's a good candidate for std::vector.